docs/eventing/channel-based-broker.md

```diff
@@ -0,0 +1,214 @@
+event routing. You will need to have a Channel provider installed, for example
+InMemoryChannel (for development purposes), Kafka, Nats, etc. You can choose from
+list of [available channels](https://knative.dev/docs/eventing/channels/channels-crds/)
+
+Once you have decided which Channel(s) you want to use and have installed them, you
+can configure the Broker by controlling which Channel(s) are used. You can choose
+this as a cluster level default, by namespace or by a specific Broker. These are
+configured by a `config-br-defaults` `ConfigMap` in knative-eventing namespace.
+
+Here's an example of a configuration that uses Kafka channel for all the
+Brokers except namespace `test-broker-6` which uses InMemoryChannels. First
+define the `ConfigMap`s to describe how the Channels of each type are created:
+
+```yaml
+# Define how InMemoryChannels are created
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: knative-eventing
+  name: imc-channel
+data:
+  channelTemplateSpec.apiVersion: messaging.knative.dev/v1alpha1
+  channelTemplateSpec.kind: InMemoryChannel
+```
+
+```yaml
+# Define how Kafka channels are created. Note we specify
+# extra parameters that are particular to Kakfa Channels, namely
+# numPartitions as well as replicationFactor.
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kafka-channel
+  namespace: knative-eventing
+data:
+  channelTemplateSpec.apiVersion: messaging.knative.dev/v1alpha1
+  channelTemplateSpec.kind: KafkaChannel
+  channelTemplateSpec.specJson: '{"numPartitions": 3, "replicationFactor": 1}'
+```
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: config-br-defaults
+  namespace: knative-eventing
+data:
+  default-br-config: |
+    clusterDefault:
+      apiVersion: v1
+      kind: ConfigMap
+      name: imc-channel
+      namespace: knative-eventing
+    namespaceDefaults:
+      test-broker-6:
+        apiVersion: v1
+        kind: ConfigMap
+        name: kafka-channel
+        namespace: knative-eventing
+```
+
+
+## Installing Broker by Annotation
+
+The easiest way to get Broker installed, is to annotate your namespace
+(replace `default` with the desired namespace):
+
+```shell
+kubectl label namespace default knative-eventing-injection=enabled
+```
+
+This will automatically create a `Broker` named `default` in the `default`
+namespace. As per above configuration, it would be configured to use Kafka
+channels.
+
+```shell
+kubectl -n default get broker default
+```
+
+_NOTE_ `Broker`s created due to annotation will not be removed if you remove the
+annotation. For example, if you annotate the namespace, which will then create
+the `Broker` as described above. If you now remove the annotation, the `Broker`
+will not be removed, you have to manually delete it.
+
+For example, to delete the injected Broker from the foo namespace:
+
+```shell
+kubectl -n foo delete broker default
+```
+
+## Installing Broker Manually
```

Comment:
Honestly I think this section should be removed. Why would a user ever do this if they can annotate the namespace (and if they can create ServiceAccounts, they can probably annotate the namespace)? Maybe a contributor needs to know, but then it can be part of the eventing repo.

Reply:
Yeah, I initially thought about removing it entirely, but for the cases where you might have multiple types of brokers in a given namespace, you can't just label the namespace. You could for example have a Broker foo and bar, and you would need to create one of them manually.